middleware/kubernetes/k8sclient/dataobjects.go

```diff
@@ -5,9 +5,16 @@ import (
     "net/http"
 )
 
+// getK8sAPIResponse wraps the http.Get(url) function to provide dependency
+// injection for unit testing.
+var getK8sAPIResponse = func(url string) (resp *http.Response, err error) {
+	resp, err = http.Get(url)
+	return resp, err
+}
+
 
-func getJson(url string, target interface{}) error {
-    r, err := http.Get(url)
+func parseJson(url string, target interface{}) error {
```

Comment:
target is not used in this function.

Why do we need his extra level of indirection?